src/Android/Services/AutofillHandler.cs

```diff
@@ -71,9 +71,17 @@ public bool SupportsAutofillService()
             }
         }
 
-        public void Autofill(CipherView cipher)
+        public void Autofill(CipherView cipher, bool isExternalSelection = false)
         {
-            var activity = (MainActivity)CrossCurrentActivity.Current.Activity;
+            Xamarin.Forms.Platform.Android.FormsAppCompatActivity activity;
+            if (isExternalSelection)
+            {
+                activity = (AutofillExternalSelectionActivity)CrossCurrentActivity.Current.Activity;
+            }
+            else
+            {
+                activity = (MainActivity)CrossCurrentActivity.Current.Activity;
+            }
```

Comment:
Just in case, if the current activity is not the one you're casting the method will throw an exception of invalid casting. Is that ok? if not you can cast using `as` that if it fails to cast, it returns `null`. However, we should log something if that happens to check why is not casting correctly and what is trying to cast.
Furthermore, if you want you can avoid the `bool isExternalSelection = false` given that the only casting you need is to `FormsAppCompatActivity`, you're not using anything from the explicit casting so the next thing should be enough:
```c#
var activity = CrossCurrentActivity.Current.Activity as Xamarin.Forms.Platform.Android.FormsAppCompatActivity;
```
Now, if you want to explicitly check whether the activity is one of those two, check it before, log it and return.